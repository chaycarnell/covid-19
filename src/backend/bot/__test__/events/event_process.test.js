const { processBody } = require('../../bot.helpers');
const {
  trade_open,
  trade_close,
  equity_change,
  balance_change
} = require('./event_mocks.js');

test('Format a open trade event', () => {
  expect(processBody({ body: trade_open })).toMatchSnapshot();
});

test('Format a close trade event', () => {
  expect(processBody({ body: trade_close })).toMatchSnapshot();
});

test('Format a equity change event', () => {
  expect(processBody({ body: equity_change })).toMatchSnapshot();
});

test('Format a balance change event', () => {
  expect(processBody({ body: balance_change })).toMatchSnapshot();
});
