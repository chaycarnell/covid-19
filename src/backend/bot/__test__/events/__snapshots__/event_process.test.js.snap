// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Format a balance change event 1`] = `
Object {
  "account": "20684668",
  "balance": "1000",
  "category": "account",
  "count": "1",
  "equity": "1100",
  "type": "balance_change",
}
`;

exports[`Format a close trade event 1`] = `
Object {
  "account": "20684668",
  "bias": "sell",
  "category": "trade",
  "close_balance": "1050",
  "close_time": "2019-10-11 19:02:54",
  "equity": "1050",
  "lots": "0.10",
  "symbol": "USDCAD",
  "ticket": "206366873",
  "total_profit": 50,
  "type": "trade_close",
}
`;

exports[`Format a equity change event 1`] = `
Object {
  "account": "20684668",
  "balance": "1500",
  "category": "account",
  "count": "1",
  "equity": "1500",
  "type": "equity_change",
}
`;

exports[`Format a open trade event 1`] = `
Object {
  "account": "20684668",
  "bias": "sell",
  "category": "trade",
  "equity": "998.82",
  "lots": "0.10",
  "open_balance": "1000",
  "open_time": "2019-10-11 18:59:11",
  "symbol": "USDCAD",
  "ticket": "206366873",
  "type": "trade_open",
}
`;
